### PR TITLE
fix: retry when connect to eventbus in sensor

### PR DESCRIPTION
**Pull request by exherb**
when lots of sensors got updated, eventbus connect is really unstable.

### CODE REVIEW

**Comment by whynowy** [comment]
We already have a retry function at https://github.com/argoproj/argo-events/blob/master/eventsources/sources/util.go#L18, can you switch to use that?

**Comment by exherb** [comment]
eventsource will crash and restart, so I think it’s ok to not retry connect when start?

**Comment by exherb** [comment]
ok, I will switch to util retry

**Comment by exherb** [comment]
> We already have a retry function at https://github.com/argoproj/argo-events/blob/master/eventsources/sources/util.go#L18, can you switch to use that?

done

File: sensors/listener.go [hunk 1]
```diff
@@ -102,9 +103,14 @@ func (sensorCtx *SensorContext) ListenEvents(ctx context.Context, stopCh <-chan
...
-			conn, err := ebDriver.Connect()
+			var conn eventbusdriver.Connection
+			err = sources.Connect(&common.DefaultRetry, func() error {
+				var err error
+				conn, err = ebDriver.Connect()
+				return err
+			})
 			if err != nil {
-				logger.Error("failed to connect to event bus", zap.Error(err))
+				logger.Fatal("failed to connect to event bus", zap.Error(err))
```
Comment (whynowy) [inline sensors/listener.go:113]: No need to change to `fatal`, the returned error will be wrapped as a `fatal` error in main.go

**Pull request merged**